cmd/kubectl-gadget/utils/trace.go

```diff
@@ -371,7 +371,15 @@ func getTraceListFromID(traceID string) (*gadgetv1alpha1.TraceList, error) {
 // SetTraceOperation sets the operation of an existing trace.
 // If trace does not exist an error is returned.
 func SetTraceOperation(traceID string, operation string) error {
-	traces, err := getTraceListFromID(traceID)
+	// We have to wait for the end of previous operation before changing the trace
+	// operation.
+	// Trace controller set Annotations to nil when it is dealing with an
+	// operation.
+	// Thus, we need to wait for Annotations to become nil (empty) before being
+	// able to change the current operation.
+	// It is the same like when you are in the restaurant, you need to wait the
+	// chef to cook the main dishes before ordering the dessert.
+	traces, err := waitForEmptyAnnotations(traceID)
```

Comment:
I think it has to be more specific here. It's not about all annotations but only for the `"gadget.kinvolk.io/operation"`. I mean, it should check that one of the following is true: `"gadget.kinvolk.io/operation"` is empty, doesn't exist at all or the whole annotations are nil. Otherwise, if we add another annotation this logic will break because they won't be nil.

Reply:
Oops! You are right, I will better check the [`GADGET_OPERATION`](https://github.com/kinvolk/inspektor-gadget/blob/main/pkg/controllers/trace_controller.go#L228)!